### PR TITLE
Gradient Accumulation

### PR DESCRIPTION
**Pull request by Sea-Snell**
I noticed that it was a requested feature in the README and I needed it for my own project, so I implemented gradient accumulation for training Bert with larger batches on a lower memory device.

### CODE REVIEW

**Comment by 652994331** [comment]
@Sea-Snell   wow, so good, that's exactly what i need. Thanks so much. However, since i am new for tensorflow, could you please show me how to use these codes?(there are many  files). Btw, i am doing a bert finetune, do i need to replace my optimization.py with yours? Or maybe i need modify more. thanks for your help, really appreciate.

**Comment by Sea-Snell** [comment]
Yes great to hear that my code is helpful! So you'll want to replace with my optimization.py file and my run_classifier.py file to run pre training, and when you call run_classifier.py you will need to add a acum_steps flag, which is how many batches to run before updating the gradients. Let me know if you have any questions!

**Comment by 652994331** [comment]
@Sea-Snell Thanks for your reply, I did not know we don't have FLAG.accum_step before, so I added accum_step = FLAGS.accum_step myself and did some tiny change(btw i was using the run_classifier.py and optimization.py from "fixed global counter coherency with learn rate decay and grad") The code is running now, it seems that the fine-tune speed is slower than normal optimization. For now I maintain accum_step = 1. However, according to my understanding, accum_step = 1 means we are not using multi-step update,so if accum_step = 1 we are actually use the normal optimization.py , but the code is running with a large batch size which cant be used before. Still trying to figure out the reason, could you please give me some advice?
